sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -993,6 +994,14 @@ class CodegenContext {
     }
   }
 
+  /**
+   * Returns the code for subexpression elimination after splitting it if necessary.
+   */
+  def subexprFunctionsCode: String = {
+    // Wholestage codegen does not allow subexpression elimination
```

Comment:
shall we add `assert(currentVars == null)` to guarantee it?

Reply:
cc @maropu , I vaguely remember that we do subexpression elimination in hash aggregate in one of your PR.

Reply:
I added it, thanks. Anyway, when we generate subexpression elimination functions they all take an `InternalRow` as input. Unless that mechanism changes, I doubt it is usable in wholestage codegen.

Reply:
> shall we add `assert(currentVars == null)` to guarantee it?

I cannot do that, it doesn't work. In that case just an empty string is returned because the subexpression functions seq is empty.

Reply:
btw, (this is a off-topic though) the `HashAggregateExec` code for common subexpr elimination has the same issue? That also expands all generated the code for CSE in a single method now;
https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala#L287

Reply:
If so, how about `assert(currentVars != null && subexprFunctions.isEmpty)` for strict checks?

Reply:
I can add that, thanks